src/App.vue

```diff
@@ -59,31 +59,95 @@
         <b-navbar-item
           v-else
           class="login"
-          :href="logout_url"
+          v-on:click="logout()"
           title="Log out from ELIXIR Beacon Network"
         >
           Log Out
         </b-navbar-item>
       </template>
     </b-navbar>
+    <div v-if="loading">
+      <div class="loading-text">Loading</div>
+      <div class="loading-indicator spinner">
+        <Loading></Loading>
+      </div>
+    </div>
     <router-view />
     <Footer />
   </div>
 </template>
 
 <script>
+import Loading from "vue-material-design-icons/Loading.vue";
 import Footer from "@/components/Footer.vue";
+import * as oauth from "@panva/oauth4webapi";
 export default {
   data() {
     return {
       login_url: process.env.VUE_APP_LOGIN_URL,
       logout_url: process.env.VUE_APP_LOGOUT_URL,
+      loading: false,
+      issuer_url: process.env.VUE_APP_ISSUER,
+      client_id: process.env.VUE_APP_CLIENT_ID,
+      client_secret: process.env.VUE_APP_CLIENT_SECRET,
+      callback_url: process.env.VUE_APP_CALLBACK,
+      state: process.env.VUE_APP_STATE,
     };
   },
   components: {
     Footer,
+    Loading,
   },
   methods: {
+    oidcAuth: async function () {
+      this.loading = true;
+      const issuer = new URL(this.issuer_url);
+      const authorizationServer = await oauth
+        .discoveryRequest(issuer)
+        .then((response) => oauth.processDiscoveryResponse(issuer, response));
+
+      const client = {
+        client_id: this.client_id,
+        client_secret: this.client_secret,
+        token_endpoint_auth_method: "client_secret_basic",
+      };
+
+      const redirect_uri = this.callback_url;
+
+      if (
+        authorizationServer.code_challenge_methods_supported?.includes(
+          "S256"
+        ) !== true
+      ) {
+        // This example assumes S256 PKCE support is signalled
+        // If it isn't supported, random `nonce` should be used for CSRF protection.
+        throw new Error();
+      }
+
+      const code_verifier = oauth.generateRandomCodeVerifier();
+      const code_challenge = await oauth.calculatePKCECodeChallenge(
+        code_verifier
+      );
+
+      sessionStorage.setItem("code_verifier", JSON.stringify(code_verifier));
+
+      const code_challenge_method = "S256";
+
+      const authorizationUrl = new URL(
+        authorizationServer.authorization_endpoint
+      );
+      authorizationUrl.searchParams.set("client_id", client.client_id);
+      authorizationUrl.searchParams.set("code_challenge", code_challenge);
+      authorizationUrl.searchParams.set(
+        "code_challenge_method",
+        code_challenge_method
+      );
+      authorizationUrl.searchParams.set("redirect_uri", redirect_uri);
+      authorizationUrl.searchParams.set("response_type", "code");
+      authorizationUrl.searchParams.set("scope", "openid ga4gh_passport_v1");
```

Comment:
get scope from `VUE_APP_SCOPE`, so we can change the scopes with environment variables instead of rebuilding the application

Reply:
Done in latest commit